File: ChatGPT_BP_PY/CWE-119_BP/119_2a_BP.py
def get_vehicle_index():
    """
    Prompts the user to enter an index and returns the index as an integer.
    Includes input validation to ensure the entered value is a valid integer.
    """
    while True:
        try:
            index = int(input("Enter the index of the vehicle: "))
            return index
        except ValueError:
            print("Invalid input. Please enter a valid integer.")

def print_vehicle_at_index(vehicles, index):
    """
    Prints the vehicle at the given index if the index is valid.
    Includes error handling to ensure the index is within the bounds of the list.
    """
    try:
        vehicle = vehicles[index]
        print(f"Vehicle at index {index}: {vehicle}")
    except IndexError:
        print("Index out of range. Please enter a valid index.")

def main():
    """
    Main function that orchestrates the user input and printing of the vehicle.
    """
    vehicles = ['Car', 'Bike', 'Truck', 'Boat', 'Plane']  # Example vehicle list
    index = get_vehicle_index()
    print_vehicle_at_index(vehicles, index)

if __name__ == "__main__":
    main()

File: GitHubCopilot_BP_Java/CWE-119/119_2a_BP.java
package GitHubCopilot_BP_Java.CWE-119;

import java.util.Scanner;

public class Main {
    private static String[] vehicles = {"Car", "Bike", "Truck", "Bus", "Scooter"};

    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        System.out.print("Enter the index of the vehicle: ");
        String input = scanner.nextLine();
        
        try {
            int index = Integer.parseInt(input);
            printVehicleAtIndex(index);
        } catch (NumberFormatException e) {
            System.out.println("Invalid input. Please enter a valid integer.");
        }
    }

    private static void printVehicleAtIndex(int index) {
        if (index >= 0 && index < vehicles.length) {
            System.out.println("Vehicle at index " + index + ": " + vehicles[index]);
        } else {
            System.out.println("Index out of bounds. Please enter a valid index between 0 and " + (vehicles.length - 1) + ".");
        }
    }
}